pytorch_lightning/plugins/collective/__init__.py

```diff
@@ -0,0 +1,18 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pytorch_lightning.plugins.collective.collective_plugin import Collective  # noqa: F401
+from pytorch_lightning.plugins.collective.horovod_collective import HorovodCollective  # noqa: F401
+from pytorch_lightning.plugins.collective.single_device_collective import SingleDeviceCollective  # noqa: F401
+from pytorch_lightning.plugins.collective.torch_collective import TorchCollective  # noqa: F401
+from pytorch_lightning.plugins.collective.tpu_collective import TPUCollective  # noqa: F401
```

Comment:
@justusschock @awaelchli @SeanNaren is there a naming convention we should follow for plugins? the style here is different from https://github.com/PyTorchLightning/pytorch-lightning/tree/master/pytorch_lightning/plugins/io

i think we should have or establish a common pattern here to avoid issues like https://github.com/PyTorchLightning/pytorch-lightning/issues/6094

Reply:
I think we don't have one (yet).

Personally, I think the base/abstract class should contain the `'Plugin'` suffix while the others should not. But that's just personal preference :)

Reply:
Yes, I believe yours is quite neat ! We could rename `torch_plugin.py` to `torch_checkpoint_plugin.py` @SeanNaren

Reply:
How about this
1. File Name: base class has *_plugin.py and subclass doesn't. eg: collective_plugin.py and torch_collective.py (+1 @justusschock's suggestion)
2. Base Class Name:  cemal case of the file name WithOut 'plugin' and With first letter capitalized eg: collective_plugin.py -> class Collective; checkpoint_io_plugin.py -> class CheckpointIO
2. Class name: cemal case of the file name With first letter capitalized. eg: torch_collective.py -> class TorchCollective; torch_checkpoint_io_plugin.py -> class TorchCheckpointIO

@ananthsub @tchaton @awaelchli @justusschock @SeanNaren

Reply:
I would add plugin to the name of the baseclass as well (just to make it easier to recognize that it is implementing a plugin interface when subclassing it).

Reply:
address @justusschock's comments
1. File Name: base class has *_plugin.py and subclass doesn't. eg: collective_plugin.py and torch_collective.py (+1 @justusschock's suggestion)
2. Class Name: cemal case of the file name With first letter capitalized eg: collective_plugin.py -> class CollectivePlugin; checkpoint_io_plugin.py -> class CheckpointIOPlugin; torch_collective.py -> class TorchCollective; torch_checkpoint_io_plugin.py -> class TorchCheckpointIO

Reply:
Sounds good :)